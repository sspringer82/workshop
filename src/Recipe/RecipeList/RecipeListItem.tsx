import React from 'react';
import { Recipe } from '../../util/recipe';
import { Link } from 'react-router-dom';

type Props = {
  recipe: Recipe;
  onDelete: (id: number) => void;
};

const RecipeListItem: React.FC<Props> = ({ recipe, onDelete }) => {
  return (
    <li className="flex justify-between odd:bg-gray-200 p-2">
      <div className="pl-4">
        <Link to={`/detail/${recipe.id}`} data-testid="title">
          {recipe.title}
        </Link>
      </div>
      <div className="pr-4">
        <button
          className="border border-gray-500 px-2 rounded"
          onClick={() => onDelete(recipe.id)}
        >
          ðŸ—‘ lÃ¶schen
        </button>
      </div>
    </li>
  );
};

export default RecipeListItem;
