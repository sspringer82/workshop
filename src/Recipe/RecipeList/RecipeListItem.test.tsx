import React from 'react';
import { render, screen } from '@testing-library/react';
import RecipeListItem from './RecipeListItem';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('RecipeListItem', () => {
  const recipe = {
    id: 1,
    title: 'Schnitzel',
    ingredients: ['1000g Fleisch'],
    steps: ['braten'],
  };

  it('should render the name of a recipe', () => {
    const { getByText } = render(
      <BrowserRouter>
        <RecipeListItem recipe={recipe} onDelete={jest.fn()} />,
      </BrowserRouter>,
    );

    expect(getByText('Schnitzel')).toBeInTheDocument();
    expect(screen.getByTestId('title')).toHaveTextContent('Schnitzel');
  });

  it('snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <RecipeListItem recipe={recipe} onDelete={jest.fn()} />,
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
