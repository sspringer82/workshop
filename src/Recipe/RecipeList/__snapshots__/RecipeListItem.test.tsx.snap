// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<li
  class="flex justify-between odd:bg-gray-200 p-2"
>
  <div
    class="pl-4"
  >
    <a
      data-testid="title"
      href="/detail/1"
    >
      Schnitzel
    </a>
  </div>
  <div
    class="pr-4"
  >
    <button
      class="border border-gray-500 px-2 rounded"
    >
      ðŸ—‘ lÃ¶schen
    </button>
  </div>
</li>
`;

exports[`RecipeListItem snapshot 1`] = `
<li
  class="flex justify-between odd:bg-gray-200 p-2"
>
  <div
    class="pl-4"
  >
    <a
      data-testid="title"
      href="/detail/1"
    >
      Schnitzel
    </a>
  </div>
  <div
    class="pr-4"
  >
    <button
      class="border border-gray-500 px-2 rounded"
    >
      ðŸ—‘ lÃ¶schen
    </button>
  </div>
</li>
`;
